pkg/helper/client.go

```diff
@@ -56,28 +56,42 @@ func applyVlanFiltering(bridgeName string, ports []string) (string, error) {
 	return stdout.String(), nil
 }
 
-func set(state string) (string, error) {
-	cmd := exec.Command(nmstateCommand, "set")
+func nmstatectl(arguments []string, input string) (string, error) {
+	cmd := exec.Command(nmstateCommand, arguments...)
 	var stdout, stderr bytes.Buffer
 	cmd.Stderr = &stderr
 	cmd.Stdout = &stdout
-	stdin, err := cmd.StdinPipe()
-	if err != nil {
-		return "", fmt.Errorf("failed to create pipe for writing into nmstate: %v", err)
-	}
-	go func() {
-		defer stdin.Close()
-		_, err = io.WriteString(stdin, state)
+	if input != "" {
+		stdin, err := cmd.StdinPipe()
 		if err != nil {
-			fmt.Printf("failed to write state into stdin: %v\n", err)
+			return "", fmt.Errorf("failed to create pipe for writing into %s: %v", nmstateCommand, err)
 		}
-	}()
+		go func() {
+			defer stdin.Close()
+			_, err = io.WriteString(stdin, input)
+			if err != nil {
+				fmt.Printf("failed to write input into stdin: %v\n", err)
+			}
+		}()
 
-	if err = cmd.Run(); err != nil {
-		return "", fmt.Errorf("failed to execute nmstate set: '%v' '%s' '%s'", err, stdout.String(), stderr.String())
 	}
-
+	if err := cmd.Run(); err != nil {
+		return "", fmt.Errorf("failed to execute %s %s: '%v' '%v' '%s'", nmstateCommand, arguments, err, stdout.String(), stderr.String())
+	}
 	return stdout.String(), nil
+
+}
+
+func set(state string) (string, error) {
+	return nmstatectl([]string{"set", "--no-commit"}, state)
+}
+
+func Rollback() (string, error) {
```

Comment:
Why is Rollback and Commit public while set and show were private?

Reply:
We will use it later on for the lose connectivity to API I was expecting to do that outside of client.go, at controller, but we can do that check at ApplyDesiredState

Reply:
I see, still, please keep it private until we will need it elsewhere. Otherwise we may just forget about it

Reply:
ack